### PR TITLE
Add B2B self service application logic

### PR DESCRIPTION
**Pull request by inthirakumaaran**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/ServerApplicationManagementService.java [hunk 1]
```diff
@@ -517,11 +518,21 @@ public String createApplication(ApplicationModel applicationModel, String templa
         ServiceProvider application = new ApiModelToServiceProvider().apply(applicationModel);
         try {
             applicationId = getApplicationManagementService().createApplication(application, tenantDomain, username);
+
+            // Update owner for B2B Self Service applications.
+            if (application.isB2BSelfServiceApp()) {
+                String systemUserID = ApplicationMgtUtil.getB2BSelfServiceSystemUser(tenantDomain);
```
Comment (inthirakumaaran) [inline components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/ServerApplicationManagementService.java:524]: Need to move this logic to organization mgt-core

File: pom.xml [hunk 2]
```diff
@@ -572,6 +578,12 @@
         <!--<maven.checkstyleplugin.excludes>**/gen/**/*</maven.checkstyleplugin.excludes>-->
 
         <tenant.resource.manager.version>1.5.55</tenant.resource.manager.version>
+
+        <!-- Organization management core Version -->
+        <org.wso2.carbon.identity.organization.management.core.version>1.0.34
```
Comment (inthirakumaaran) [inline pom.xml:583]: Need to bump this version

File: components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/ServerApplicationManagementService.java (hunk 1 above)
Comment (inthirakumaaran) [reply components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/ServerApplicationManagementService.java]: done

**Review by thanujalk — APPROVED** [review]

File: components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/gen/java/org/wso2/carbon/identity/api/server/application/management/v1/ApplicationModel.java [hunk 3]
```diff
@@ -179,6 +180,22 @@ public void setIsManagementApp(Boolean isManagementApp) {
...
+    public ApplicationModel isB2BSelfServiceApp(Boolean isB2BSelfServiceApp) {
+
+        this.isB2BSelfServiceApp = isB2BSelfServiceApp;
+        return this;
+    }
+
+    @ApiModelProperty(example = "false", value = "Decides whether the application used to for B2B self service")
+    @JsonProperty("isB2BSelfServiceApp")
+    @Valid
+    public Boolean getIsB2BSelfServiceApp() {
```
Comment (inthirakumaaran) [inline components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/gen/java/org/wso2/carbon/identity/api/server/application/management/v1/ApplicationModel.java:195]:
```suggestion
    public Boolean getIsB2BSelfServiceApp() {
    
```

File: pom.xml (hunk 2 above)
Comment (inthirakumaaran) [reply pom.xml:583]: Need to bump framework version as well

**Review by thanujalk — APPROVED** [review]

**Pull request merged**